tensorflow_lite_support/ios/task/audio/core/audio_record/sources/TFLAudioRecord.m

```diff
@@ -0,0 +1,269 @@
+// Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "tensorflow_lite_support/ios/task/audio/core/audio_record/sources/TFLAudioRecord.h"
+
+#import "tensorflow_lite_support/ios/sources/TFLCommonUtils.h"
+#import "tensorflow_lite_support/ios/task/audio/core/audio_record/sources/TFLAudioRecordErrorCode.h"
+#import "tensorflow_lite_support/ios/task/audio/core/sources/TFLRingBuffer.h"
+
+#import <AVFoundation/AVFoundation.h>
+
+#define SUPPORTED_CHANNEL_COUNT 2
+
+/** Error domain for TFL Audio Record Errors. */
+static NSString *const TFLAudioRecordErrorDomain = @"org.tensorflow.lite.audio.record";
+
+@implementation TFLAudioRecord {
+  AVAudioEngine *_audioEngine;
+
+  /* Specifying a custom buffer size on AVAUdioEngine while tapping does not take effect. Hence we
+   * are storing the returned samples in a ring buffer to acheive the desired buffer size. If
+   * specified buffer size is shorter than the buffer size supported by `AVAUdioEngine` only the most
+   * recent data of the buffer of size `bufferSize` will be stored by the ring buffer. */
+  TFLRingBuffer *_ringBuffer;
+  dispatch_queue_t _conversionQueue;
+  NSError *_globalError;
+}
+
+- (nullable instancetype)initWithAudioFormat:(TFLAudioFormat *)audioFormat
+                                  bufferSize:(NSUInteger)bufferSize
+                                       error:(NSError *_Nullable *)error {
+  self = [self init];
+  if (self) {
+    if (audioFormat.channelCount > SUPPORTED_CHANNEL_COUNT) {
+      [TFLCommonUtils
+          createCustomError:error
+                 withDomain:TFLAudioRecordErrorDomain
+                       code:TFLAudioRecordErrorCodeInvalidArgumentError
+                description:@"The channel count provided does not match the supported "
+                            @"channel count. Only upto 2 audio channels are currently supported."];
```

Comment:
nit: As you defined SUPPORTED_CHANNEL_COUNT as a macro, it's better to use the macro value here instead of hard-code it in the error string.